006-throttle-decorator.js

```diff
@@ -0,0 +1,19 @@
+function throttle(fn, period) {
```

Comment:
consider of using

function throttle(callback, limit, context = this) {
  let inThrottle = false;
  let callbackArgs = null;

  const later = () => callback.apply(context, callbackArgs);

  return function () {
    if (!inThrottle) {
      callbackArgs = arguments;
      later();
      inThrottle = true;
      
      return setTimeout(() => inThrottle = false, limit);
    }
  };
}

const throttled = throttle(() => { console.log('throttle') }, 2000);

// These will log 'throttle' 1 time after 2 seconds delay
throttled();
throttled();
throttled();
throttled();
throttled();